### PR TITLE
Post comment by review

### PR DESCRIPTION
**Pull request by john-kieran-robson**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -136,3 +136,52 @@ describe("GET /api/reviews/:review_id/comments", () => {
...
+      .post("/api/reviews/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body.comment).toMatchObject({
+          votes: 0,
+          author: newComment.username,
+          review_id: 1,
+          created_at: expect.any(String),
+        });
```
Comment (MKD01) [inline __tests__/app.test.js:165]: body and username should also be within this .toMatchObject

File: __tests__/app.test.js [hunk 2]
```diff
@@ -169,28 +169,41 @@ describe("POST /api/reviews/:review_id/comments ", () => {
...
-  // });
+  test("status 400: user does not exist", () => {
+    const newComment = {
+      username: "hghsdfghsdlfjghlsdfkjgh",
+      body: "hello",
+    };
+    return supertest(app)
+      .post("/api/reviews/1/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (MKD01) [inline __tests__/app.test.js:204]: This should be a 404 as that username is still valid but simply does not exist.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -169,28 +169,41 @@ describe("POST /api/reviews/:review_id/comments ", () => {
...
+    const newComment = {
+      username: "mallionaire",
+      body: "ehl",
+    };
+    return supertest(app)
+      .post("/api/reviews/73/comments")
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("invalid ID");
```
Comment (MKD01) [inline __tests__/app.test.js:192]: In this case the id is valid but it simply does not exist, which also leads the question what if this was invalid? would that need to be a separate test?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Please do add those changes and submit another PR to reflect this.

File: __tests__/app.test.js (hunk 1 above)
Comment (john-kieran-robson) [reply __tests__/app.test.js:165]: user name is author will update body now

**Pull request merged**